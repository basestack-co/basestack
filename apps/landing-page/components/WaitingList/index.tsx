import React, { useEffect, useState, useRef } from "react";
import Link from "next/link";
// Styles
import { rem } from "polished";
import { useTheme } from "styled-components";
// Form
import { useForm, Controller, SubmitHandler } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
// Utils
import { z } from "zod";
import { useMedia } from "react-use";
import toast from "react-hot-toast";
// Components
import Image from "../Image";
import { Button, ButtonSize, Text } from "@basestack/design-system";
import {
  Container,
  ContentContainer,
  ImageContainer,
  TextContainer,
  Input,
  InputContainer,
  Header,
  Grid,
  Footer,
  InputWrapper,
  LeftCol,
  CardsContainer,
  ErrorContainer,
  ErrorText,
  CardWrapper,
} from "./styles";
import SlideCard from "../SlideCard";

interface WaitingListProps {
  data: Array<{
    icon: string;
    title: string;
    text: string;
    image: { src: string; alt: string };
  }>;
}

export const FormSchema = z.object({
  email: z
    .string()
    .min(1, "Required field")
    .email("This is not a valid email."),
});

export type FormInputs = z.TypeOf<typeof FormSchema>;

const WaitingList = ({ data }: WaitingListProps) => {
  const theme = useTheme();
  const isMobile = useMedia(theme.device.max.md, false);
  const [currentImage, setCurrentImage] = useState(0);
  const cardRef = useRef<HTMLDivElement>(null);

  const image = {
    src: data[currentImage].image.src,
    alt: data[currentImage].image.alt,
  };

  useEffect(() => {
    if (cardRef.current && isMobile) {
      cardRef.current.scrollIntoView({
        behavior: "smooth",
        inline: "start",
        block: "nearest",
      });
    }
  }, [cardRef, currentImage, isMobile]);

  useEffect(() => {
    const intervalId = setInterval(() => {
      const nextIndex = (currentImage + 1) % data.length;
      setCurrentImage(nextIndex);
    }, 10000);
    return () => clearInterval(intervalId);
  }, [currentImage, data]);

  const {
    control,
    handleSubmit,
    formState: { errors, isSubmitting },
    reset,
  } = useForm<FormInputs>({
    resolver: zodResolver(FormSchema),
    mode: "onChange",
  });

  const onSubmit: SubmitHandler<FormInputs> = async (input: FormInputs) => {
    try {
      const params = new URLSearchParams({
        includeListId: "4", // Feature Flags Early Access
        email: input.email,
        // templateId: "1", // Default Template Double opt-in confirmation
        // redirectionUrl,
      });

      const res = await fetch(
        `${
          process.env.NEXT_PUBLIC_EMAIL_SERVICE
        }/contacts/create?${params.toString()}`,
        {
          method: "POST",
          headers: {
            Accept: "application/json",
            "Content-Type": "application/json",
          },
        },
      );

      const data = await res.json();

      if (data.code === "permission_denied" || data.error) {
        throw new Error(data.message);
      }

      // @ts-ignore
      umami.track("request-access-submit", {
        title: "Success",
        description: "Successfully added to the waiting list",
      });

      toast.success("Thank you for your interest! We will be in touch soon.");
      reset();
    } catch (error) {
      const { message } = error as Error;
      // @ts-ignore
      umami.track("request-access-submit", {
        title: "Error",
        description: message,
      });
      toast.error(message ?? "Something went wrong, please try again.");
    }
  };

  return (
    <Container>
      <ContentContainer>
        <Header>
          <Text
            size="xxLarge"
            fontWeight={700}
            lineHeight="1.6"
            fontFamily="robotoFlex"
          >
            Basestack
          </Text>
        </Header>

        <TextContainer>
          <Text
            size="xxLarge"
            fontSize={rem("60px")}
            lineHeight="1.2"
            fontFamily="robotoFlex"
            // @ts-ignore
            as="h1"
            color={theme.colors.black}
            mb={theme.spacing.s5}
            fontWeight={800}
          >
            The Open-Source Stack for Developers and Startups
          </Text>
          <Text
            size="xxLarge"
            fontWeight={400}
            lineHeight="1.6"
            // @ts-ignore
            as="p"
            color={theme.colors.gray500}
          >
            Feature Flags are just the beginning: Our Suite of Tools will
            include Feedback, Forms, and more to help you build better Products!
          </Text>
        </TextContainer>

        <Grid>
          <LeftCol>
            <InputContainer>
              <InputWrapper>
                <Controller
                  name="email"
                  control={control}
                  defaultValue=""
                  render={({ field }) => (
                    <Input
                      placeholder="Enter your email"
                      type="email"
                      name={field.name}
                      value={field.value}
                      onChange={field.onChange}
                      onBlur={field.onBlur}
                      disabled={isSubmitting}
                      onFocus={() =>
                        // @ts-ignore
                        umami.track("email-input-focus", {
                          title: "onFocus",
                          description: "Email Input Field Focus",
                        })
                      }
                    />
                  )}
                />
                <Button
                  onClick={handleSubmit(onSubmit)}
                  isDisabled={isSubmitting}
                  isLoading={isSubmitting}
                  size={ButtonSize.Medium}
                  flexShrink={0}
                  {...(isSubmitting ? {} : { icon: "arrow_forward" })}
                >
                  Get Early Access
                </Button>
              </InputWrapper>

              <ErrorContainer>
                <ErrorText
                  size="small"
                  color={
                    !!errors.email ? theme.colors.red400 : theme.colors.gray500
                  }
                >
                  {errors.email?.message || "Be an early adopter!"}
                </ErrorText>
              </ErrorContainer>

              {!!errors.email?.message && (
                <ErrorContainer>
                  <ErrorText
                    size="small"
                    color={
                      !!errors.email
                        ? theme.colors.red400
                        : theme.colors.gray500
                    }
                  >
                    {errors.email?.message}
                  </ErrorText>
                </ErrorContainer>
              )}
            </InputContainer>
            <CardsContainer>
              {data?.map((item, index) => (
                <CardWrapper
                  key={index}
                  ref={index === currentImage ? cardRef : null}
                >
                  <SlideCard
                    isActive={index === currentImage}
                    icon={item.icon}
                    title={item.title}
                    text={item.text}
                    onClick={() => {
                      // @ts-ignore
                      umami.track("slider-card-button", {
                        title: "Feature Slider",
                        description: item.title,
                      });

                      setCurrentImage(index);
                    }}
                  />
                </CardWrapper>
              ))}
            </CardsContainer>
          </LeftCol>
          <ImageContainer>
            <Image src={image.src} alt={image.alt} />
          </ImageContainer>
        </Grid>

        <Footer>
          <Text size="medium" fontWeight={400} muted>
            Â© Basestack {new Date().getFullYear()}. All rights reserved.{" "}
            <Link
              style={{ color: "black" }}
              href="/legal/privacy"
              onClick={() =>
                // @ts-ignore
                umami.track("privacy-link", {
                  title: "onClick",
                  description: "Privacy Policy Link",
                })
              }
            >
              Privacy Policy
            </Link>
          </Text>
        </Footer>
      </ContentContainer>
    </Container>
  );
};

export default WaitingList;
