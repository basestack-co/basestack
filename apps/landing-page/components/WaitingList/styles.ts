import styled, { css } from "styled-components";
import { Text } from "@basestack/design-system";
import { rem } from "polished";

const flexColumn = css`
  display: flex;
  flex-direction: column;
`;

export const Container = styled.section`
  ${flexColumn};
  min-height: 100vh;
  position: relative;
  z-index: 1;
  overflow: hidden;

  &::before {
    content: "";
    background: rgba(238, 238, 238, 0.7);
    -webkit-backdrop-filter: saturate(180%) blur(150px);
    backdrop-filter: saturate(180%) blur(150px);
    position: absolute;
    z-index: -1;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
  }
`;

export const ContentContainer = styled.div`
  ${flexColumn};
  flex-grow: 1;
  max-width: ${rem("1440px")};
  width: 100%;
  margin: 0 auto;
`;

export const Header = styled.div`
  ${flexColumn};
  padding: ${({ theme }) => theme.spacing.s8} ${({ theme }) => theme.spacing.s5}
    0 ${({ theme }) => theme.spacing.s5};
`;

export const TextContainer = styled.div`
  ${flexColumn};
  max-width: 50%;
  padding: ${({ theme }) => theme.spacing.s8} ${({ theme }) => theme.spacing.s5};

  @media screen and ${({ theme }) => theme.device.max.xl} {
    max-width: 80%;
  }

  @media screen and ${({ theme }) => theme.device.max.lg} {
    max-width: 100%;
  }

  @media screen and ${({ theme }) => theme.device.max.md} {
    h1 {
      font-size: 42px;
    }

    p {
      font-size: 20px;
    }
  }

  @media screen and ${({ theme }) => theme.device.max.sm} {
    h1 {
      font-size: 32px;
    }

    p {
      font-size: 18px;
    }
  }
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: minmax(auto, ${rem("420px")}) 1fr;
  grid-gap: ${({ theme }) => theme.spacing.s3};

  @media screen and ${({ theme }) => theme.device.max.lg} {
    grid-template-columns: 1fr;
  }

  @media screen and ${({ theme }) => theme.device.max.lg} {
    grid-gap: ${({ theme }) => theme.spacing.s6};
  }
`;

export const LeftCol = styled.div`
  ${flexColumn};
`;

export const InputContainer = styled.div`
  ${flexColumn};
  padding: 0 ${({ theme }) => theme.spacing.s5};
`;

export const InputWrapper = styled.div`
  display: flex;
`;

export const ImageContainer = styled.div`
  ${flexColumn};
  padding: 0 ${({ theme }) => theme.spacing.s5};
  position: relative;

  &::after {
    content: "";
    background: ${({ theme }) => theme.colors.primary};
    opacity: 0.5;
    position: absolute;
    border-radius: 50%;
    z-index: -2;
    height: 500px;
    width: 500px;
    right: 20px;
    top: -150px;
  }
`;

export const Input = styled.input`
  font-size: ${rem("16px")};
  height: ${rem("42px")};
  width: 100%;
  max-width: ${rem("320px")};
  border-radius: ${rem("4px")};
  margin-right: ${({ theme }) => theme.spacing.s3};
  padding: 0 ${({ theme }) => theme.spacing.s4};
  color: ${({ theme }) => theme.colors.gray900};
  border: 2px solid ${({ theme }) => theme.colors.gray600};
  background-color: transparent;

  &::placeholder {
    color: ${({ theme }) => theme.colors.gray500};
  }

  &:focus {
    outline: 2px solid ${({ theme }) => theme.colors.gray300};
  }
`;

export const ErrorContainer = styled.div`
  height: 0;
  position: relative;
`;

export const ErrorText = styled(Text)`
  position: absolute;
  top: ${rem("4px")};
  left: 0;
`;

export const CardsContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.spacing.s5};
  margin-top: ${({ theme }) => theme.spacing.s6};
  padding: ${({ theme }) => theme.spacing.s5};

  @media screen and ${({ theme }) => theme.device.max.lg} {
    flex-direction: row;
  }

  @media screen and ${({ theme }) => theme.device.max.md} {
    width: 100vw;
    overflow-x: scroll;
    scroll-snap-type: x mandatory;
    scroll-padding: ${({ theme }) => theme.spacing.s5};

    -ms-overflow-style: none;
    scrollbar-width: none;

    &::-webkit-scrollbar {
      display: none;
    }
  }
`;

export const CardWrapper = styled.div`
  display: flex;
  flex: 1 0 0;

  @media screen and ${({ theme }) => theme.device.max.md} {
    min-width: 320px;
    scroll-snap-align: start;
  }
`;

export const Footer = styled.div`
  ${flexColumn};
  padding: ${({ theme }) => theme.spacing.s8} ${({ theme }) => theme.spacing.s5};
  margin-top: auto;
`;
