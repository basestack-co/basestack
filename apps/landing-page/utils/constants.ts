export const sendinblueApiUrl = "https://api.sendinblue.com/v3";
export const redirectionUrl = "https://basestack-landing-page.onrender.com/";
