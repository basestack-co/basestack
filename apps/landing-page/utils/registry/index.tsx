"use client";

import Footer from "components/Footer";
import GlobalNavigation from "components/GlobalNavigation";
import { usePathname } from "next/navigation";
import React, { Fragment, useMemo } from "react";
// Components
import { Toaster } from "sonner";
// Registries
import StyledComponentsRegistry from "./StyledComponentsRegistry";
// Fonts
import "material-symbols/rounded.css";

const Registry = ({ children }: { children: React.ReactNode }) => {
  const pathname = usePathname();
  const isProductPage = useMemo(
    () => pathname.includes("/product"),
    [pathname],
  );

  const hasGlobalNavigation = useMemo(
    () =>
      !["/legal/privacy", "/legal/cookies", "/legal/terms"].includes(pathname),
    [pathname],
  );

  return (
    <StyledComponentsRegistry>
      <Fragment>
        {hasGlobalNavigation && <GlobalNavigation isSticky={!isProductPage} />}
        {children}
        <Footer />
      </Fragment>
      <Toaster visibleToasts={9} closeButton={false} />
    </StyledComponentsRegistry>
  );
};

export default Registry;
