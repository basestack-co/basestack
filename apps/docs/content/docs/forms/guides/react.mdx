---
title: NextJS & React
icon: Code
tags: Basestack Forms, NextJS, React Integration, Form submission, Fetch API, Axios, React, Next.js, API integration, JSON response, Web development, Developer tools, "2025"
---

import { Step, Steps } from 'fumadocs-ui/components/steps';

[React](https://react.dev/?ref=basestack.co) is a JavaScript library for building user interfaces, allowing developers to create interactive, dynamic web applications with reusable components. [Next.js](https://nextjs.org/?ref=basestack.co) is a React framework that enables server-side rendering, static site generation, and simplified routing for building fast, scalable web applications.

To get started, youâ€™ll need the following:

- A [Basestack Forms](https://forms.basestack.co) Account
- A React or Next.js project with a Form component

<Steps>

<Step>
### Use this on your React Project to send form data to your endpoint.

Send data using the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API?ref=basestack.co), [Axios](https://axios-http.com/?ref=basestack.co) or Any [HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP?ref=basestack.co) Client and ensure the mode on the query string is set to `mode=rest` for retrieving the response in JSON format.
</Step>
<Step>
### Copy and paste this into your React component.

Feel free to use the code below to send a test submission.


```tsx copy title="contact.tsx"
import React, { useState } from "react";

const FormComponent = () => {
  // You can also use rect-hook-form or formik for form handling
  // check out the docs for more info https://react-hook-form.com/
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  const onHandleSubmit = (e) => {
    e.preventDefault();
    e.stopPropagation();

    // You can also use axios or any other library to make the request
    // You can also use async await to make the request
    fetch("https://forms.basestack.co/api/v1/s/[KEY]?mode=rest", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ name, email, message }),
    })
      .then((res) => res.json())
      .then((data) => {
        if (data.code === 200) {
          console.log("Form submitted successfully");
        }
      })
      .catch((error) => {
        console.log("Error submitting form", error);
      });
  };

  return (
    <form onSubmit={(e) => onHandleSubmit(e)}>
      <div>
        <label htmlFor="name">Name:</label>
        {/*  <!-- name each of your inputs as you wish --> */}
        <input
          type="text"
          id="name"
          name="name"
          required
          autoComplete="name"
          aria-label="Name"
          placeholder="Enter your name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </div>

      <div>
        <label htmlFor="email">Email:</label>
        <input
          type="email"
          id="email"
          name="email"
          required
          autoComplete="email"
          aria-label="Email"
          placeholder="Enter your email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </div>

      <div>
        <label htmlFor="message">Message:</label>
        <textarea
          id="message"
          name="message"
          required
          aria-label="Message"
          placeholder="Enter your message"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
        ></textarea>
      </div>

      {/*  <!-- your other form fields go here --> */}

      <button type="submit">Submit</button>
    </form>
  );
};

export default FormComponent;
```
</Step>
</Steps>
