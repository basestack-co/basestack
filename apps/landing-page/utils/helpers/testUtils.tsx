import theme from "@basestack/design-system/theme/lightTheme";
// Styles
import isPropValid from "@emotion/is-prop-valid";
import { RenderOptions, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

interface WithChildrenProps {
  children: React.ReactNode;
}

interface AllProvidersProps extends WithChildrenProps {
  initialState?: {};
}

const AllProviders = ({ children, initialState = {} }: AllProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const renderWithAllProviders = (
  ui: React.ReactElement,
  initialState?: {},
  options?: Omit<RenderOptions, "wrapper">,
) =>
  render(ui, {
    // @ts-ignore
    wrapper: (props) => <AllProviders initialState={initialState} {...props} />,
    ...options,
  });

const WithThemeProvider = ({ children }: WithChildrenProps) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  // @ts-ignore
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithAllProviders, renderWithTheme };
