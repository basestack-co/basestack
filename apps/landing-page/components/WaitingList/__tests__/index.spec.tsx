import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import WaitingList from "..";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => {
    return {
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  }),
});

describe("WaitingList tests", () => {
  afterEach(cleanup);

  test("should render WaitingList correctly", () => {
    const { asFragment } = renderWithTheme(
      <WaitingList
        data={[
          {
            icon: "help",
            title: "title",
            text: "text",
            image: { src: "", alt: "" },
          },
        ]}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
