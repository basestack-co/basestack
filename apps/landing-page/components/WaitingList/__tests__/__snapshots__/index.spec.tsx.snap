// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaitingList tests should render WaitingList correctly 1`] = `
<DocumentFragment>
  .c27 {
  display: flex;
  flex-direction: column;
  background-color: #F6F6F6;
  box-shadow: rgba(0, 0, 0, 0.25) 0px 25px 50px -12px;
  border-radius: 0.5rem;
  width: 100%;
  overflow: hidden;
}

.c28 {
  width: 100%;
  height: auto;
}

.c16 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: roboto;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #757575;
}

.c25 {
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 500;
  font-family: roboto;
  font-weight: 400;
  line-height: 1.6;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #757575;
}

.c30 {
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 500;
  font-family: roboto;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #757575;
}

.c24 {
  font-size: 1.125rem;
  line-height: 1.625rem;
  font-weight: 500;
  font-family: roboto;
  margin-left: 0.75rem;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #000000;
}

.c3 {
  font-size: 1.5rem;
  line-height: 2.25rem;
  font-weight: 700;
  font-family: robotoFlex;
  font-weight: 700;
  line-height: 1.6;
  font-family: 'Roboto Flex',sans-serif;
  word-break: break-word;
  color: #000000;
}

.c5 {
  font-size: 1.5rem;
  line-height: 2.25rem;
  font-weight: 700;
  font-family: robotoFlex;
  font-size: 3.75rem;
  line-height: 1.2;
  margin-bottom: 1.25rem;
  font-weight: 800;
  font-family: 'Roboto Flex',sans-serif;
  word-break: break-word;
  color: #000000;
}

.c6 {
  font-size: 1.5rem;
  line-height: 2.25rem;
  font-weight: 700;
  font-family: roboto;
  font-weight: 400;
  line-height: 1.6;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #757575;
}

.c13 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
  margin-left: 0.5rem;
}

.c22 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c14 {
  color: #000000;
  font-size: 1.5rem;
}

.c23 {
  color: #276EF1;
  font-size: 1.5rem;
}

.c12 {
  height: 2.625rem;
  padding: 0 1rem;
  font-size: 1rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #276EF1;
  color: #FFFFFF;
  width: medium;
  height: medium;
  flex-shrink: 0;
  width: fit-content;
}

.c12:hover:not(:active, :disabled) {
  background-color: #0e55d7;
}

.c12 .material-symbols-sharp {
  color: inherit;
}

.c12 svg,
.c12 svg g {
  fill: #FFFFFF;
}

.c12:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c0 {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  position: relative;
  z-index: 1;
  overflow: hidden;
}

.c0::before {
  content: "";
  background: rgba(238, 238, 238, 0.7);
  -webkit-backdrop-filter: saturate(180%) blur(150px);
  backdrop-filter: saturate(180%) blur(150px);
  position: absolute;
  z-index: -1;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
}

.c1 {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  max-width: 90rem;
  width: 100%;
  margin: 0 auto;
}

.c2 {
  display: flex;
  flex-direction: column;
  padding: 3.125rem 1.25rem 0 1.25rem;
}

.c4 {
  display: flex;
  flex-direction: column;
  max-width: 50%;
  padding: 3.125rem 1.25rem;
}

.c7 {
  display: grid;
  grid-template-columns: minmax(auto, 26.25rem) 1fr;
  grid-gap: 0.75rem;
}

.c8 {
  display: flex;
  flex-direction: column;
}

.c9 {
  display: flex;
  flex-direction: column;
  padding: 0 1.25rem;
}

.c10 {
  display: flex;
}

.c26 {
  display: flex;
  flex-direction: column;
  padding: 0 1.25rem;
  position: relative;
}

.c26::after {
  content: "";
  background: #276EF1;
  opacity: 0.5;
  position: absolute;
  border-radius: 50%;
  z-index: -2;
  height: 500px;
  width: 500px;
  right: 20px;
  top: -150px;
}

.c11 {
  font-size: 1rem;
  height: 2.625rem;
  width: 100%;
  max-width: 20rem;
  border-radius: 0.25rem;
  margin-right: 0.75rem;
  padding: 0 1rem;
  color: #141414;
  border: 2px solid #545454;
  background-color: transparent;
}

.c11::placeholder {
  color: #757575;
}

.c11:focus {
  outline: 2px solid #CBCBCB;
}

.c15 {
  height: 0;
  position: relative;
}

.c17 {
  position: absolute;
  top: 0.25rem;
  left: 0;
}

.c18 {
  display: flex;
  flex-direction: column;
  gap: 1.25rem;
  margin-top: 1.875rem;
  padding: 1.25rem;
}

.c19 {
  display: flex;
  flex: 1 0 0;
}

.c29 {
  display: flex;
  flex-direction: column;
  padding: 3.125rem 1.25rem;
  margin-top: auto;
}

.c20 {
  position: relative;
  border: none;
  display: flex;
  flex-direction: column;
  text-align: left;
  background-color: #FFFFFF;
  border-radius: 0.5rem;
  padding: 1.25rem 1.25rem 1.5625rem 1.25rem;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 6px -1px,rgba(0, 0, 0, 0.06) 0px 2px 4px -1px;
  cursor: pointer;
  transition: scale 0.2s ease-in-out;
  overflow: hidden;
}

.c20::before {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  animation: bHTXJE 10s linear;
  height: 5px;
  background-color: #276EF1;
}

.c21 {
  display: flex;
  align-items: center;
  margin-bottom: 0.75rem;
}

@media screen and (max-width: 1199.98px) {
  .c4 {
    max-width: 80%;
  }
}

@media screen and (max-width: 991.98px) {
  .c4 {
    max-width: 100%;
  }
}

@media screen and (max-width: 767.98px) {
  .c4 h1 {
    font-size: 42px;
  }

  .c4 p {
    font-size: 20px;
  }
}

@media screen and (max-width: 575.98px) {
  .c4 h1 {
    font-size: 32px;
  }

  .c4 p {
    font-size: 18px;
  }
}

@media screen and (max-width: 991.98px) {
  .c7 {
    grid-template-columns: 1fr;
  }
}

@media screen and (max-width: 991.98px) {
  .c7 {
    grid-gap: 1.875rem;
  }
}

@media screen and (max-width: 991.98px) {
  .c18 {
    flex-direction: row;
  }
}

@media screen and (max-width: 767.98px) {
  .c18 {
    width: 100vw;
    overflow-x: scroll;
    scroll-snap-type: x mandatory;
    scroll-padding: 1.25rem;
    -ms-overflow-style: none;
    scrollbar-width: none;
  }

  .c18::-webkit-scrollbar {
    display: none;
  }
}

@media screen and (max-width: 767.98px) {
  .c19 {
    min-width: 320px;
    scroll-snap-align: start;
  }
}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
          font-family="robotoFlex"
          font-weight="700"
        >
          Basestack
        </h1>
      </div>
      <div
        class="c4"
      >
        <h1
          class="c5"
          color="#000000"
          font-family="robotoFlex"
          font-size="3.75rem"
          font-weight="800"
        >
          The Open-Source Stack for Developers and Startups
        </h1>
        <p
          class="c6"
          color="#757575"
          font-family="roboto"
          font-weight="400"
        >
          Feature Flags are just the beginning: Our Suite of Tools will include Feedback, Forms, and more to help you build better Products!
        </p>
      </div>
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              <input
                class="c11"
                name="email"
                placeholder="Enter your email"
                type="email"
                value=""
              />
              <button
                class="c12"
                data-testid="button"
              >
                <span
                  class=""
                >
                  Get Early Access
                </span>
                <div
                  class="c13"
                  data-testid="icon-container"
                  size="24"
                >
                  <span
                    class="c14 material-symbols-sharp"
                    data-testid="icon"
                    size="24"
                  >
                    arrow_forward
                  </span>
                </div>
              </button>
            </div>
            <div
              class="c15"
            >
              <p
                class="c16 c17"
                color="#757575"
                font-family="roboto"
              >
                Be an early adopter!
              </p>
            </div>
          </div>
          <div
            class="c18"
          >
            <div
              class="c19"
            >
              <button
                class="c20"
              >
                <div
                  class="c21"
                >
                  <div
                    class="c22"
                    data-testid="icon-container"
                    size="24"
                  >
                    <span
                      class="c23 material-symbols-sharp"
                      color="#276EF1"
                      data-testid="icon"
                      size="24"
                    >
                      help
                    </span>
                  </div>
                  <h3
                    class="c24"
                    font-family="roboto"
                  >
                    title
                  </h3>
                </div>
                <p
                  class="c25"
                  font-family="roboto"
                  font-weight="400"
                >
                  text
                </p>
              </button>
            </div>
          </div>
        </div>
        <div
          class="c26"
        >
          <div
            class="c27"
          >
            <img
              alt=""
              class="c28"
              src=""
            />
          </div>
        </div>
      </div>
      <div
        class="c29"
      >
        <p
          class="c30"
          font-family="roboto"
          font-weight="400"
        >
          © Basestack 2024. All rights reserved. 
          <a
            href="/legal/privacy"
            style="color: black;"
          >
            Privacy Policy
          </a>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
